### DISCUSSION TITLE
Is anyone using Beanie with AWS DocumentDB?

### DISCUSSION THREAD

**Discussion by eze1981** [BeanieODM/beanie#441] [Question] [answered]
Hi all,

I have a FastAPI that I developed using a local mongo db instance, but when deploying the production environment in AWS using DocumentDB, I get the following error message: `pymongo.errors.OperationFailure: Field 'collation' is currently not supported`.

The error occurs during the application initialization, during the execution of "init_beanie":

```python
@app.on_event("startup")
async def startup_event():
    request_id = str(uuid.uuid4())
    logger.info(f"App startup. Request [{request_id}]].")
    await init_beanie(
        database=db,
        document_models=[
            User,
        ],
    )
```

Then, this is my definition for _User_ model:

```python
class User(BeanieBaseUser[PydanticObjectId]):
    first_name: Optional[str]
    last_name: Optional[str]
    affiliation: Optional[str]
    studies: list = []
    pass
```

Has anyone else had success using beani with AWS DocumentDB? My DocumentDB cluster is version 4.0. I understand that Amazon's implementation of the MongoDB API is incomplete, but I'm not sure if this error is because of that or maybe it is something else.

I'll appreciate it if you could share your experience with beani on AWS DocumentDB.

Thank you

**Comment by eze1981** [comment, marked as answer]
Hi all,

I solved this issue by overriding the BeaniBaseUser internal Settings class. Here you have an example to remove the collation configuration:

```python
class User(BeanieBaseUser[PydanticObjectId]):
    first_name: Optional[str]
    last_name: Optional[str]
    affiliation: Optional[str]
    studies: list = []
    
    class Settings:
        email_collation = None
        indexes = [
            IndexModel("email", unique=True)
        ]

    pass
```

**Reply by amirsalaar** [reply]
I have had the same issue. DocumentDB does not support the Collation feature unfortunately. 
@eze1981 how heavy are you using Beanie in production? I am going to use DocumentDB as my db as well but I am not sure if Beanie would work with it...

**Reply by eze1981** [reply]
Hi @amirsalaar, I'm sorry for my late answer. I was providing support to a customer who ended up migrating to MongoAtlasDB. I don't have any workload in production under my control.
